x-pack/plugins/proxy/syncing.md

```diff
@@ -0,0 +1,169 @@
+# Client Syncing Strategy
```

Comment:
Yes! This should have been written _before_ any of this code was written (and would have really streamlined the process -- and really should be the path we are following BEFORE implementation). However, you deal with what you have.

I am leaving it _in_ this PR because it describes, in plain language, how this current sync strategy evolved.